core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -855,6 +855,76 @@ private SqlNode callToSql(@Nullable RexProgram program, RexCall call0,
       final RexCall call = (RexCall) stripCastFromString(call1, dialect);
       SqlOperator op = call.getOperator();
       switch (op.getKind()) {
+      case IS_FALSE:
```

Comment:
Can this be done in unparseCall instead?
Then you don't need the new policy function.

Reply:
If we use unparseCall, we need to parse the SqlCall. We can't extract whether it is deterministic from SqlCall. So, I put it here.

Reply:
true.
I don't really like this function "supportsUnparseOnlyDeterministic()", because it really has nothing to do with IS_TRUE, etc. But I don't have a better proposal on how to structure this.

Reply:
I have also been pondering over this for a long time. How can I accurately express it so that the developers can quickly figure out which functions should be added here and which ones should use unparseCall instead?
How about `unparseCallPartial` The meaning is that this function "unparse" is supported in certain scenarios, not in all cases.

Reply:
The most generic way to handle this would be to add a hook in the dialect to rewrite calls before they are converted to sql. This hook would be called from callToSql prior to converting to SqlNode.

Reply:
I understand. I will try to do it this way.

Reply:
the only misgivings I have about that is that the Dialect class is mostly about SqlNode.
But it does have a method that takes a RexNode

Reply:
Adding RexNode parsing to Dialect might render the current unparseCall ineffective. After all, RexNode offers greater flexibility.

Reply:
Or we could change "supportsUnparseOnlyDeterministic" in the current PR to "supportsUnparsePartial" and mark it as @Experimental. Then we'll see if it's necessary to add the parsing of RexNode to Dialect later.

Reply:
I am thinking of a function in Dialect like this: `RexCall prepareUnparse(RexCall arg)` which can rewrite a RexCall before it is unparsed; the default implementation returns arg. The argument could be RexNode too.